dashboard/test/lib/email_reminder_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+require 'email_reminder'
+
+class EmailReminderTest < ActiveSupport::TestCase
+  test 'finds a permission request that needs a reminder' do
+    student = create :student
+    student.update(child_account_compliance_state: 'not_g')
+    request = ParentalPermissionRequest.create(user_id: student.id, parent_email: 'foo-parent@code.org')
```

Comment:
I think you could extract the above 3 lines and the repeated 3 lines in teach of the tests into a `setup_all` block at the top of the class. It will run the block once, before the tests run, and set up your user. I think you'll have to change the variables to instance variables (@myvar) to access them in the unit tests, but it would save a lot of repeated set up code.

Example:
```
class EmailReminderTest < ActiveSupport::TestCase
  setup_all do
    @student = create :student
    @student.update(child_account_compliance_state: 'not_g')
    @request = ParentalPermissionRequest.create(user_id: @student.id, parent_email: 'foo-parent@code.org')
  end

  # then your unit tests below
```

Reply:
There's also `setup`, which I think runs before each test, vs `setup_all` which runs once for all the tests.